notebooks/cookbook/Enhanced sampling methods/Umbrella sampling MD.ipynb

```diff
@@ -0,0 +1,436 @@
+    "prd_steps = 100     "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "5"
+      ]
+     },
+     "execution_count": 5,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "## Load an already solvated PDB file and set up the system + state\n",
+    "pdb = PDBFile(receptor_path)\n",
+    "omm_forcefield = ForceField(\"amber/ff14SB.xml\", \"amber14/tip3p.xml\")\n",
+    "system = omm_forcefield.createSystem(pdb.topology, \n",
+    "                                         nonbondedMethod=PME, \n",
+    "                                         nonbondedCutoff=nb_cutoff, \n",
+    "                                         constraints=HBonds, \n",
+    "                                         rigidWater=True,\n",
+    "                                         hydrogenMass=H_mass)\n",
+    "system.addForce(MonteCarloBarostat(1*bar, sim_temp))\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "prot_top = md.Topology.from_openmm(pdb.topology)\n",
+    "# find three atoms and save their inds - these will be used to define distances\n",
+    "d1_atom1_ind = prot_top.select(\"residue 6 and name CZ\")[0]\n",
+    "d1_atom2_ind = prot_top.select(\"residue 10 and name CZ\")[0] \n",
+    "d2_atom2_ind = prot_top.select(\"residue 17 and name CZ\")[0]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "1"
+      ]
+     },
+     "execution_count": 7,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# Add Harmonica functions to CV bonds\n",
+    "bias_d1 = CustomBondForce(\"0.5*k*(r-r0_d1)^2\")\n",
+    "bias_d1.addGlobalParameter(\"k\", 1.0)\n",
+    "bias_d1.addGlobalParameter(\"r0_d1\", 0.3)\n",
+    "\n",
+    "bias_d2 = CustomBondForce(\"0.5*k*(r-r0_d2)^2\")\n",
+    "bias_d2.addGlobalParameter(\"k\", 1.0)\n",
+    "bias_d2.addGlobalParameter(\"r0_d2\", 0.3)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "7"
+      ]
+     },
+     "execution_count": 8,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# Add forces to the system\n",
+    "bias_d1.addBond(d1_atom1_ind, d1_atom2_ind)\n",
+    "bias_d2.addBond(d1_atom1_ind, d2_atom2_ind)\n",
+    "\n",
+    "system.addForce(bias_d1)\n",
+    "system.addForce(bias_d2)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Define an integrator where the harmonic boost potential applied is written out as a global variable"
```

Comment:
I have no clue what this CustomIntegrator is for.  It isn't needed for umbrella sampling.  Just use LangevinMiddleIntegrator.